### ISSUE TITLE
Problems with 2 examples in  'ARIA required context role' (ff89c9)

### ISSUE THREAD

**Issue by dd8** [act-rules/act-rules.github.io#1174]
Passed example 2 causes problems in some AT:

- not read as a list in Safari 13 with VoiceOver on macOS 10.15 (reads list item as plain text)
- List item not announced correctly by JAWS 2019 with IE11 (reads 'list start', 'list end' without reading list item)

It fails 4.1.1 because `ul` requires `li` children which is the likely cause of the problems above.
https://act-rules.github.io/rules/ff89c9#passed-example-2

Failed Example 3 doesn't cause any problems in any AT we tested:

- voiced correctly as a a list with 1 list item in JAWS 2019.1912.1 with IE11, FF68, Chrome 80
- voiced correctly as a a list with 1 list item in NVDA 2019.2.1 with IE11, FF68, Chrome 80
- voiced correctly as a a list with 1 list item in VoiceOver on macOS 10.15.3 with Safari 13.0.5

It's not clear from any of the ARIA specs why this is a failure.
https://act-rules.github.io/rules/ff89c9#failed-example-3

**Comment by EmmaJP** [comment]
Suggestion from meeting 26/3 to make each list have at least 2 items.

**Comment by WilcoFiers** [comment]
As Emma suggested, this was discussed during today's call. Decision of the group was to leave passed example 2 as is. It properly demonstrates the rule, and while this may or may not be a failure under 4.1.1, it is not a clear failure under 1.3.1, which is what this rule is testing. We will need to add an accessibility support not about potential issues with Safari and IE.

We've also decided that, to better demonstrate the issue of failed example 3, we'll add multiple listitems. The issue here is that the listitem in failed example 3 is treated as its own list. Multiple listitems would more clearly demonstrate that the context rule isn't picked up properly. This needs to happen to other examples too.

**Comment by Jym77** [comment]
@dd8 I've tried to fix the problems as part of #1423 .
- For Passed Example 2, I added a note saying that it does not satisfy 4.1.1. Plus an Accessibility Support about some combinations of UA/AT not working well with combination of implicit/explicit roles.
- For ailed Example 3, I've added a second list item (actually to all the examples). From my quick test in Chrome+NVDA, only one of them is announced as being in a list (of 1 element) and the other out of all lists.

**Comment by dd8** [comment]
One problem with mixing ARIA and invalid HTML is the HTML parsing algorithm may re-order the HTML. For example:

```
    <h1>Cheese <h2 role='button'>Order</h2> </h1>
```

During parsing (and before ARIA is applied), the DOM is reordered into:

```
<h1>Cheese </h1>
<h2 role='button'>Order</h2>    
```

I've not found a definitive statement in the specs on what happens if you mix native roles and ARIA roles like passed example 1:

```
<ul>
	<div role="listitem">List item 1</div>
</ul>
```

Could be this is defined somewhere (and I've missed it) but if not then isn't this undefined behaviour?

**Comment by Jym77** [comment]
Good point.

I think we are fairly OK with that problem for two reasons:
1. The input aspect is the DOM tree, not the HTML source. Therefore, if the DOM tree has been somehow "fixed" by the UA, the rule will only see that fixed version. (compare for example with [Attribute is not duplicated](https://act-rules.github.io/rules/e6952f) where the input aspect is the source code because DOM tree building already fix that problem).
2. Testing in Chrome, Edge and Firefox, they all fix the `<h1><h2></h2></h1>` example the same way, but none of them change the `<ul><div></div></ul>` example (Passed Example 2). Therefore, Passed Example 2 should indeed pass on these 3 major browsers as the DOM tree they build matches the source code.
